src/main/java/build/buildfarm/instance/stub/StubInstance.java

```diff
@@ -853,4 +867,12 @@ public void deregisterWorker(String workerName) {
             .deregisterWorker(
                 DeregisterWorkerRequest.newBuilder().setWorkerName(workerName).build());
   }
+
+  @Override
+  public DrainWorkerPipelineResults drainWorkerPipeline(String worker) {
+    throwIfStopped();
+    return shutDownWorkerGracefullyBlockingStub
+        .get()
+        .drainWorkerPipeline(DrainWorkerPipelineRequest.newBuilder().build());
```

Comment:
I think `worker` needs added the the `DrainWorkerPipelineRequest`

Reply:
The worker ip:port is used to create the ManagedChannel, so we are reaching that worker directly. Do you think we should do something with the `worker`?